pkg/sink/sink.go

```diff
@@ -150,6 +112,63 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 		r.EventListenerName, r.EventListenerNamespace, string(eventID), string(event), request.Header)
 }
 
+func (r Sink) processTrigger(t *triggersv1.EventListenerTrigger, request *http.Request, event []byte, eventID string, eventLog *zap.SugaredLogger) error {
+	if t == nil {
+		return errors.New("EventListenerTrigger not defined")
+	}
+	log := eventLog.With(zap.String(triggersv1.TriggerLabelKey, t.Name))
+
+	finalPayload, err := r.executeInterceptors(t, request, event, eventID, log)
+	if err != nil {
+		return err
+	}
+
+	rt, err := template.ResolveTrigger(*t,
+		r.TriggersClient.TektonV1alpha1().TriggerBindings(r.EventListenerNamespace).Get,
+		r.TriggersClient.TektonV1alpha1().TriggerTemplates(r.EventListenerNamespace).Get)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	params, err := template.ResolveParams(rt.TriggerBindings, finalPayload, request.Header, rt.TriggerTemplate.Spec.Params)
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	log.Info("params: %+v", params)
+	resources := template.ResolveResources(rt.TriggerTemplate, params)
+	if err := r.createResources(resources, t.Name, eventID); err != nil {
+		log.Error(err)
+		return err
```

Comment:
to make it a bit easier to trace through errors, could we wrap the error with a bit of context here? e.g. fmt.Errorf("something useful: %w", err)